oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakRBACAuthorizer.java

```diff
@@ -405,17 +436,20 @@ public List<AuthorizationResult> authorize(AuthorizableRequestContext requestCon
             }
 
             if (grants != null) {
-                return allowOrDenyBasedOnGrants(requestContext, actions, grants);
+                result = allowOrDenyBasedOnGrants(requestContext, actions, grants);
+            } else {
+                result = delegateIfRequested(requestContext, actions, null);
             }
-
-            return delegateIfRequested(requestContext, actions, null);
+            addAuthzMetricSuccessTime(startTime);
+            return result;
 
         } catch (Throwable t) {
```

Comment:
We should not catch and swallow `Throwable`

Reply:
Can you give an example where this would be really bad here?

In the case of any kind of failure here the authorization will be denied, and full stacktrace will be logged.
In this case a custom authorizer is actually not a separate piece without which the rest of the broker can function. If authorizer doesn't function, the broker doesn't function.

I guess by re-throwing an Error if it occurs that would signal Kafka to shutdown. Is that the desired behaviour in this case? Often times, especially on Kubernetes, it is easier to debug an issue without causing the JVM to shutdown.
But then what good is the system that is up, but not working as it should due to misconfiguration or limited resources.